torch/distributions/kl.py

```diff
@@ -121,10 +128,244 @@ def kl_divergence(p, q):
+        return -p.entropy() + (q.high - q.low).log()
+    else:
+        return torch.new(p.rate.size()).fill_(float('inf'))
+
+@register_kl(Exponential, Beta)
+def _kl_exponential_beta(p, q):
+    return torch.new(p.rate.size()).fill_(float('inf'))
+
+@register_kl(Exponential, Gamma)
+def _kl_exponential_gamma(p, q):
+    t1 = q.alpha * q.beta.log() - torch.lgamma(q.alpha)
+    t2 = (q.alpha - 1) * (p.rate.log() + 0.57721566490153286060) / p.rate
+    t3 = q.beta / p.rate
+    return t1 - t2 - t3
+
+@register_kl(Exponential, Normal)
+def _kl_exponential_normal(p, q):
+    var_normal = q.std.pow(2)
+    rate_sqr = p.rate.pow(2)
+    t1 = 0.5 * torch.log(rate_sqr * var_normal * 2 * math.pi)
+    t2 = rate_sqr.reciprocal()
+    t3 = q.mean / p.rate
+    t4 = q.mean.pow(2) * 0.5
+    return t1 - 1 + (t2 - t3 + t4) / var_normal
+
+@register_kl(Exponential, Pareto)
+def _kl_exponential_pareto(p, q):
+    return torch.new(p.rate.size()).fill_(float('inf'))
+
+@register_kl(Exponential, Uniform)
+def _kl_exponential_uniform(p, q):
+    return torch.new(p.rate.size()).fill_(float('inf'))
+
+@register_kl(Gamma, Beta)
+def _kl_gamma_beta(p, q):
+    return torch.new(p.alpha.size()).fill_(float('inf'))
+
+@register_kl(Gamma, Exponential)
+def _kl_gamma_exponential(p, q):
+    return -p.entropy() - q.rate.log() + q.rate * p.alpha / p.beta
+
+@register_kl(Gamma, Normal)
+def _kl_gamma_exponential(p, q):
+    var_normal = q.std.pow(2)
+    beta_sqr = p.beta.pow(2)
+    t1 = 0.5 * torch.log(beta_sqr * var_normal * 2 * math.pi) - p.alpha - p.alpha.lgamma()
+    t2 = 0.5 * (p.alpha.pow(2) + p.alpha) / beta_sqr
+    t3 = q.mean * p.alpha / p.beta
+    t4 = 0.5 * q.mean.pow(2)
+    return t1 + (p.alpha - 1) * p.alpha.digamma() + (t2 - t3 + t4) / var_normal
+
+@register_kl(Gamma, Pareto)
+def _kl_gamma_pareto(p, q):
+    return torch.new(p.alpha.size()).fill_(float('inf'))
+
+@register_kl(Gamma, Uniform)
+def _kl_gamma_uniform(p, q):
+    return torch.new(p.alpha.size()).fill_(float('inf'))
+
+@register_kl(Laplace, Beta)
+def _kl_laplace_beta(p, q):
+    return torch.new(p.loc.size()).fill_(float('inf'))
+
+@register_kl(Laplace, Exponential)
+def _kl_laplace_exponential(p, q):
+    return torch.new(p.loc.size()).fill_(float('inf'))
+
+@register_kl(Laplace, Gamma)
+def _kl_laplace_gamma(p, q):
+    return torch.new(p.loc.size()).fill_(float('inf'))
+
+@register_kl(Laplace, Normal)
+def _kl_laplace_normal(p, q):
+    var_normal = q.std.pow(2)
+    scale_sqr_var_ratio = p.scale.pow(2) / var_normal
+    t1 = 0.5 * torch.log(2 * scale_sqr_var_ratio / math.pi)
+    t2 = 0.5 * p.loc.pow(2)
+    t3 = p.loc * q.mean
+    t4 = q.mean.pow(2)
+    return -t1 + scale_sqr_var_ratio + (t2 - t3 + t4) / var_normal - 1
+
+@register_kl(Laplace, Uniform)
+def _kl_laplace_uniform(p, q):
+    return torch.new(p.loc.size()).fill_(float('inf'))
+
+@register_kl(Normal, Beta)
+def _kl_normal_beta(p, q):
```

Comment:
I guess you could compress some of these as
```py
@register_kl(Normal, Beta)
@register_kl(Normal, Exponential)
@register_kl(Normal, Gamma)
@register_kl(Normal, Uniform)
def _kl_normal_infinite(p, q):
    return p.mean.new([float('inf')]).expand_as(p.mean)
```
Note also that this uses `p.mean.new()` rather than `torch.new()` so that the result has correct GPU placement, and uses `new([...]).expand_as(...)` which is cheaper than `new(shape).fill_(...)`.

Reply:
I combined `_infinite_like` with this to get:
```python
def _kl_normal_infinite(p, q):
    return _infinite_like(p.mean)
```